storageClient/storageClient.go

```diff
@@ -83,6 +83,18 @@ func (c *StorageClient) DeleteGroup(ctx context.Context, params DeleteGroupReque
 	return err
 }
 
+// ListGroups returns all the groups a client is associated with (if any).
+func (c *StorageClient) ListGroups(ctx context.Context, params ListGroupsRequest) (*ListGroupsResponse, error) {
+	var result *ListGroupsResponse
+	path := c.Host + storageServiceBasePath + "/groups"
+	request, err := e3dbClients.CreateRequest("GET", path, params)
```

Comment:
Get should have nil for params

Reply:
Encode the next token and max as optional url query params

Encode the client_id as an optional url query param, that way be default when you call list groups you get all groups for your account, but you can optionally have the server filter out / only return groups for the requested client id

Examples of passing query params
https://github.com/tozny/e3db-clients-go/blob/master/identityClient/identityClient.go#L202
https://github.com/tozny/e3db-clients-go/blob/master/pdsClient/pdsClient.go#L310